xarray/core/ops.py

```diff
@@ -344,11 +349,30 @@ def prod(values, axis=None, skipna=None, **kwargs):
             raise NotImplementedError(
                 'skipna=True not yet implemented for prod with dtype %s'
                 % values.dtype)
-        _fail_on_dask_array_input_skipna(values)
-        return npcompat.nanprod(values, axis=axis, **kwargs)
+        return _nanprod(values, axis=axis, **kwargs)
     return _prod(values, axis=axis, **kwargs)
 prod.numeric_only = True
 
+def cumsum(values, axis=None, skipna=None, **kwargs):
```

Comment:
Better than that-- I extended functionality to `_create_nan_agg_method` to generalize, which removes quite a few lines of code.  See `d1c1077`.